app/src/main/java/com/asfoundation/wallet/tokenswap/SwapEtherToTokenActivity.java

```diff
@@ -0,0 +1,239 @@
+package com.asfoundation.wallet.tokenswap;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+import com.asf.wallet.R;
+import dagger.android.AndroidInjection;
+import javax.inject.Inject;
+
+public class SwapEtherToTokenActivity extends AppCompatActivity
+    implements SwapView, AdapterView.OnItemSelectedListener {
+  @Inject SwapInteractor swapInteractor;
+  private EditText etSrcAmount;
+  private ResponseListener<String> showTxHash;
+  private SwapPresenter presenter;
+  private EditText amountFromView;
+  private EditText amountToView;
+  private Spinner sTokenFrom;
+  private Spinner sTokenTo;
+  private TextView tvShowRates;
+  private final String from = "FROM";
```

Comment:
instead of using this string, consider using an enum with the possible action defined on presenter